code/modules/mob/living/silicon/robot/robot_movement.dm

```diff
@@ -1,3 +1,22 @@
+/mob/living/silicon/robot/Move()
+	. = ..()
+	if(module.chop_on_move)
+		var/mob/living/carbon/M = locate(/mob/living/carbon) in loc
+		handle_slice(M)
+
+/mob/living/silicon/robot/proc/handle_slice(mob/living/carbon/M)

```

Comment:
You should probably turn all this into a component, so we can turn Ian into a lawnmower aswell

Reply:
I dunno how to do that lol

Reply:
That would be preferred as it can be on the module and not all robots.